scalding-commons/src/main/scala/com/twitter/scalding/commons/thrift/TBaseOrderedBufferable.scala

```diff
@@ -0,0 +1,83 @@
+package com.twitter.scalding.commons.thrift
+
+import com.twitter.scalding.typed.OrderedBufferable
+import cascading.tuple.hadoop.io.BufferedInputStream
+import cascading.tuple.StreamComparator
+import org.apache.thrift.protocol.TCompactProtocol
+import org.apache.thrift.protocol.TBinaryProtocol
+import org.apache.thrift.TBase
+import java.nio.ByteBuffer
+import com.twitter.bijection.Inversion.attempt
+import com.twitter.bijection.Bufferable
+import org.apache.thrift.transport.TIOStreamTransport
+
+case class ArrayOffset(toInt: Int) extends AnyVal {
+  def +(o: LocalOffset) = TotalOffset(toInt + o.toInt)
+}
+// offset within the current byte buffer
+case class LocalOffset(toInt: Int) extends AnyVal
+// offset within the array byte
+case class TotalOffset(toInt: Int) extends AnyVal
+case class Offset(toInt: Int) extends AnyVal
+case class Length(toInt: Int) extends AnyVal
+case class Remaining(toInt: Int) extends AnyVal
+case class ArrayBuf(toArrayBytes: Array[Byte]) extends AnyVal
+
+abstract class TBaseOrderedBufferable[T <: TBase[_, _]] extends OrderedBufferable[T] {
+  val minFieldId: Short
+
+  @transient private lazy val factory = new TBinaryProtocol.Factory //  new TCompactProtocol.Factory
+
+  @transient protected def prototype: T
+
+  def hash(t: T) = t.hashCode
+
+  def compare(a: T, b: T) = if (a == b) 0 else -1
```

Comment:
can we add a BIG comment here explaining this. I'd love to see a TODO to make it follow the contract of compare == compareBinary